### ISSUE TITLE
How does YOLOv5 handle gt boxes with a large aspect ratio difference

### ISSUE THREAD

**Issue by blueskywwc** [ultralytics/yolov5#1815]
In the training data, the aspect ratio of the gt box of some categories exceeds 10:1, which is a long bar. It is difficult for the trained model to predict this box. What should I do, thank you!

**Comment by glenn-jocher** [comment]
@blueskywwc the YOLOv5 dataloader has label candidate criteria that are applied on the data during training (including aspect ratio thresholds of `1/20 < ar < 20` ), and separately during NMS as well. You should review these defaults for your custom dataset. The locations are:

Dataloader
https://github.com/ultralytics/yolov5/blob/7180b22e2d2d06be1622cbc71e758ed04727ff65/utils/datasets.py#L814-L820

NMS
https://github.com/ultralytics/yolov5/blob/7180b22e2d2d06be1622cbc71e758ed04727ff65/utils/general.py#L276-L283

**Comment by blueskywwc** [comment]
thanks,I will try and update
